### PR TITLE
Nicolas/abs 450 abstract ibc setup helpers should connect both ways

### PR DESCRIPTION
**Pull request by Kayanski**
This PR aims at making the Polytone and Abstract Interchain Apis better.
Depends on https://github.com/AbstractSDK/polytone/pull/5

### Checklist

- [x] CI is green.
- [x] Changelog updated.

### CODE REVIEW

**Review by Buckram123 — APPROVED** [review]
Amazing! Is it possible to connect 3 or more abstracts?

**Comment by Kayanski** [comment]
> Amazing! Is it possible to connect 3 or more abstracts?

yes, you can see the test passing here : 
https://github.com/AbstractSDK/abstract/pull/378/files#diff-906e27c8fa15629fbce07f20986ba7e8d2ec84506006c1241e7338a53f475a5dR161

File: framework/Cargo.toml [hunk 1]
```diff
@@ -58,9 +58,13 @@ cw-orch = { version = "0.23.0" }
...
 tokio = { version = "1.4", features = ["full"] }
 
-polytone = { package = "abstract-polytone", version = "1.0.5" }
-polytone-note = { package = "abstract-polytone-note", version = "1.0.5" }
-cw-orch-polytone = { package = "abstract-cw-orch-polytone", version = "2.0.0" }
+# polytone = { package = "abstract-polytone", version = "1.0.5" }
+# polytone-note = { package = "abstract-polytone-note", version = "1.0.5" }
+# cw-orch-polytone = { package = "abstract-cw-orch-polytone", version = "2.0.0" }
+
+polytone = { package = "abstract-polytone", git = "https://github.com/abstractsdk/polytone", branch = "update/better-api" }
```
Comment (CyberHoward) [inline framework/Cargo.toml:65]: Should be versioned package (or will block us from publishing)

File: framework/packages/abstract-client/src/client.rs [hunk 2]
```diff
@@ -358,7 +358,8 @@ impl<Chain: CwEnv> AbstractClient<Chain> {
 
     #[cfg(feature = "interchain")]
     /// Connect this abstract client to the remote abstract client
-    /// It assumes that [`cw_orch_polytone::Polytone`] is deployed
+    /// If [`cw_orch_polytone::Polytone`] is deployed between 2 chains, it will NOT redeploy it (good for actual chains)
+    /// If Polytone is not deployed, deploys it between the 2 chains (good for integration testing)
     pub fn ibc_connection_with(
```
Comment (CyberHoward) [inline framework/packages/abstract-client/src/client.rs:363]: isn't it assumed that we use IBC for this?

`ibc_connection_with` -> `connect_with`?

Comment (CyberHoward) [reply framework/packages/abstract-client/src/client.rs:363]: * seems like this fn just passes is through. It should have the same fn name as the underlying fn (`connect`). Tho IMO both should be `connect_to`

**Comment by CyberHoward** [comment]
@Kayanski the interchain tests are failing locally but succeeding in CI: https://github.com/AbstractSDK/abstract/actions/runs/9682371932/job/26715226845

How does the polytone crate figure out if it should deploy? Giving me the following err: 
image

**Comment by CyberHoward** [comment]
ok just ran cargo update and now it passes!

**Comment by Buckram123** [comment]
> ok just ran cargo update and now it passes!

Git deps are evil

**Comment by Kayanski** [comment]
> > ok just ran cargo update and now it passes!
> 
> Git deps are evil

This was not git dep related, this was a patch update on my part on cw-orch-core

**Pull request merged**